nntrainer/utils/meson.build

```diff
@@ -24,9 +24,13 @@ if get_option('enable-trace')
   util_headers += 'tracer.h'
 endif
 
-if get_option('enable-fp16')
+arch = target_machine.cpu_family()
+
+if get_option('enable-fp16') && arch == 'aarch64'
   util_sources += 'util_simd_neon.cpp'
   util_headers += 'util_simd_neon.h'
+elif arch == 'arm'
+  error ('FP16/ARM code (blas_neon.cpp) uses armv8.2 instructions. armv7 is not supported.')
```

Comment:
You always get an error in arm with this code.

```
if get_option('enable-fp16') 
  if arch == 'aarch64'
    util_sources += 'util_simd_neon.cpp'
    util_headers += 'util_simd_neon.h'
  elif arch == 'arm'
    error ('FP16/ARM code (blas_neon.cpp) uses armv8.2 instructions. armv7 is not supported.')
  endif
```

Reply:
Yes, I am aware of it. (fixed in my local)
But the main problem is that `arch` during `package_android_build.sh` is `x86_64`.
codes like:
```meson
if arch == 'aarch64'
```
will not solve the current build fail.

So basically for `enable-fp16`:
```psuedo-code
if enable-fp16
  if enable-neon
    if arch == 'arm'
      error ( ... )
    else
      // add neon src files ?
```
In current android-build use, there is no case for `arch` to have `aarch64`, unless we build it from ARM based controller

Reply:
Try "host machine" instead of "target machine", too.

Target machine == aarch64 means that the compiler you have built here will create aarch64 binary. (you are not compiling a compiler): https://mesonbuild.com/Cross-compilation.html

Reply:
it still says `x86_64`.
Think it is because we have been differentiating it with `-Dplatform=android`, and actually cross-compile with ndk-build.
Will check further ASAP

Reply:
1. test host-machine, target-machine in GBS. see if it gives the correct arch name
2. if 1 == correct, it means host-machine in Android build is the only issue.
   then you can do: (psuedo code)
```
if ANDROID
  build a dummy a.c.
  run "file a.out"
  read arch name from above
  set arch.
endif
```

Reply:
And your code above is ok, too.

Reply:
Got it. Will let you know after trying that

Reply:
I checked that I could get correct  host-machine, target-machine conditions depending on `-A {$architecture}` option when GBS build.